src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -841,31 +858,35 @@
     <Field Name="Label" Type="LabelSymbol" Null="disallow"/>
   </Node>
 
-  <AbstractNode Name="BoundDecisionDag" Base="BoundNode">
+  <Node Name="BoundDecisionDag" Base="BoundNode">
+    <Field Name="RootNode" Type="BoundDecisionDagNode" Null="disallow"/>
+  </Node>
+
+  <AbstractNode Name="BoundDecisionDagNode " Base="BoundNode">
   </AbstractNode>
   <!-- This node is used to indicate a point in the decision dag where an evaluation is performed, such as invoking Deconstruct. -->
-  <Node Name="BoundEvaluationPoint" Base="BoundDecisionDag">
+  <Node Name="BoundEvaluationDecisionDagNode" Base="BoundDecisionDagNode ">
     <Field Name="Evaluation" Type="BoundDagEvaluation" Null="disallow"/>
-    <Field Name="Next" Type="BoundDecisionDag" Null="disallow"/>
+    <Field Name="Next" Type="BoundDecisionDagNode " Null="disallow"/>
   </Node>
   <!-- This node is used to indicate a point in the decision dag where a test may change the path of the decision dag. -->
-  <Node Name="BoundDecisionPoint" Base="BoundDecisionDag">
-    <Field Name="Decision" Type="BoundDagDecision" Null="disallow"/>
-    <Field Name="WhenTrue" Type="BoundDecisionDag" Null="disallow"/>
-    <Field Name="WhenFalse" Type="BoundDecisionDag" Null="disallow"/>
+  <Node Name="BoundTestDecisionDagNode" Base="BoundDecisionDagNode ">
+    <Field Name="Test" Type="BoundDagTest" Null="disallow"/>
+    <Field Name="WhenTrue" Type="BoundDecisionDagNode " Null="disallow"/>
+    <Field Name="WhenFalse" Type="BoundDecisionDagNode " Null="disallow"/>
   </Node>
-  <Node Name="BoundWhenClause" Base="BoundDecisionDag">
-    <Field Name="Bindings" Type="ImmutableArray&lt;(BoundExpression,BoundDagTemp)&gt;" Null="disallow"/>
+  <Node Name="BoundWhenDecisionDagNode" Base="BoundDecisionDagNode ">
+    <Field Name="Bindings" Type="ImmutableArray&lt;(BoundExpression left,BoundDagTemp dagTemp)&gt;" Null="disallow"/>
```

Comment:
>(BoundExpression left,BoundDagTemp dagTemp) [](start = 51, length = 43)

Please consider using uppercase for first character of tuple fields.

Reply:
I prefer to always use lowercase for tuple field names.